src/lib.rs

```diff
@@ -913,6 +913,42 @@ fn lexicographical_topological_sort(
     Ok(PyList::new(py, out_list).into())
 }
 
+#[pyfunction]
+fn graph_greedy_color(
+    py: Python,
+    graph: &graph::PyGraph,
+) -> PyResult<PyObject> {
+    let mut colors: HashMap<usize, usize> = HashMap::new();
+    let mut node_vec: Vec<NodeIndex> = graph.graph.node_indices().collect();
+    node_vec.sort_by_key(|k| graph.graph.edges(*k).count());
```

Comment:
Python networkx seems to apply stable sort with reverse mode enabled. This retworkx implementation applies a stable sort then reverse it. So, the resulting coloring of retworkx will be OK, but it may be different from that by networkx.
How about `sort_by_key(|k| {num nodes} - {num neighbors of k})`?

Reply:
Fixed in: https://github.com/Qiskit/retworkx/pull/77/commits/03b7872b67379325fd2a1ac86ed9bbec662f0810 although I just read up a bit more on the `sort_by_key` method and it looks like I can use https://doc.rust-lang.org/std/cmp/struct.Reverse.html to reverse the sort without having to add a subtraction to the key func. I'll give that a try and if it's faster I'll switch to that.

Reply:
Didn't behave as expected the results differed and it wasn't any faster, so lets just stick with what's in https://github.com/Qiskit/retworkx/commit/03b7872b67379325fd2a1ac86ed9bbec662f0810